be2-scala/docs/README.md

```diff
@@ -210,47 +224,59 @@ final case class Read(channel: Channel, id: Hash) extends Event
 `DbActor` will then answer using one of its predetermined answers defined within the same file:
 
 ```scala
-// DbActor Events correspond to messages the actor may emit
+// DbActor DbActorMessages correspond to messages the actor may emit
 sealed trait DbActorMessage
 
-final case class DbActorWriteAck() extends DbActorMessage
+final case class DbActorAck() extends DbActorMessage
 final case class DbActorReadAck(message: Option[Message]) extends DbActorMessage
 final case class DbActorCatchupAck(messages: List[Message]) extends DbActorMessage
-final case class DbActorNAck(code: Int, description: String) extends DbActorMessage
 ```
 
-Here's an example (shamefully stolen from `MessageHandler.scala`) showing the power of `DbActor` coupled with Scala [Future](https://www.scala-lang.org/files/archive/api/2.13.1/scala/concurrent/Future.html)
+When the `DbActor` fails to complete an operation – instead of sending a `DbActorMessage` –, the actor will send back to the origin a wrapped `DbActorNAckException` (containing both the error code and reason the exception occurred). The wrapper itself is a [`Status.Failure`](https://doc.akka.io/japi/akka/current/akka/actor/Status.Failure.html) often used by akka to notify a different actor that something went wrong.
+
+:information_source: Note that the `Status.Failure` takes a `Throwable` at construction. You should **not** throw the exception inside the Failure unless you feel the urge to waste 3 hours debugging an incomprehensible actor state (definitely did not go through this :smile_cat:...)!
+
+As a simplified example, we can look at the Write event from `DbActor`, where we use `this.synchronized` to avoid concurrency issues for sequential read/writes that should be seen as a single transaction. In this examples, we are writing both the message on the channel (line 5) and the message_id in the ChannelData object (stored at 'channel', line 4).
```

Comment:
`As a simplified example` of what? 
If the point here is to give an example of a read/update/write pattern, I would make the intention clearer.